### ISSUE TITLE
Type problem with Bot

### ISSUE THREAD

**Issue by LiberaTeMetuMortis** [PrismarineJS/mineflayer#3084]
- [x ] The [FAQ](https://github.com/PrismarineJS/mineflayer/blob/master/docs/FAQ.md) doesn't contain a resolution to my issue 

## Your current code
I have an variable called "event" and it is type is `keyof BotEvents`
```ts
// I have an variable called "event" and it is type is `keyof mineflayer.BotEvents`
bot.on(event, (...args) => {
// It says args is any[] but we know that it is parameter of any of BotEvents
})
```

To fix this we can change type definitions.
Current type definition is:
```ts
on<E extends keyof Events> (event: E, listener: Events[E]): this
```
but if we change it with 
```ts
on<E extends keyof Events>(event: E, listener: (...args: Parameters<Events[E]>) => unknown): this
```
the problem gets solved, now TypeScript knows args is any of the BotEvents

**Comment by extremeheat** [comment]
Why are are the consequences/side effects for your proposed fix? You can open a PR for it

**Comment by LiberaTeMetuMortis** [comment]
> What are are the consequences/side effects for your proposed fix? You can open a PR for it

There is no side effects, this is what libraries like djs use on their Event Emitter typing but the problem is this code is from an abondened library which is not updated for 2 years

**Comment by extremeheat** [comment]
Try using plain JavaScript

**Comment by lkwilson** [comment]
I don't think this is related to mineflayer. The type definition of on is in https://github.com/andywer/typed-emitter

**Comment by LiberaTeMetuMortis** [comment]
I also have an issue there but the library is no more getting updated
